src/pattern-library/components/PlaygroundApp.js

```diff
@@ -39,49 +41,129 @@ export default function PlaygroundApp({
     </>
   );
 
-  const routeGroups = [
-    { title: 'Foundations', routes: getRoutes('foundations') },
-    { title: 'Patterns', routes: getRoutes('patterns') },
-    { title: 'Common Components', routes: getRoutes('components') },
-  ];
+  /**
+   * Header for a primary section of nav
+   *
+   * @param {object} props
+   *   @param {Children} props.children
+   */
+  function NavHeader({ children }) {
+    return (
+      <h2 className="border-b px-2 py-1 font-light text-lg">{children}</h2>
+    );
+  }
 
-  if (extraRoutes.length) {
-    routeGroups.push({ title: 'Application Patterns', routes: extraRoutes });
+  /**
+   * A single navigation link
+   *
+   * @param {object} props
+   *   @param {string} props.title
+   *   @param {PlaygroundRoute['route']} props.route
```

Comment:
If you're going to tie the `route` field to the `PlaygroundRoute` type, you could just define it as `@param {PlaygroundRoute} props.route` and then get the title value from the route itself.